src/SFA.DAS.EAS.MessageHandlers.Worker/EventHandlers/ApprovedTransferConnectionInvitationEventHandler.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using SFA.DAS.EAS.Domain.Configuration;
+using SFA.DAS.EAS.Domain.Models.UserProfile;
+using SFA.DAS.EAS.Infrastructure.Data;
+using SFA.DAS.EmployerAccounts.Events.Messages;
+using SFA.DAS.Messaging;
+using SFA.DAS.Messaging.AzureServiceBus.Attributes;
+using SFA.DAS.Messaging.Interfaces;
+using SFA.DAS.NLog.Logger;
+using SFA.DAS.Notifications.Api.Client;
+using SFA.DAS.Notifications.Api.Types;
+
+namespace SFA.DAS.EAS.MessageHandlers.Worker.EventHandlers
+{
+    [TopicSubscription("Task_ApprovedTransferConnectionInvitation")]
+    public class ApprovedTransferConnectionInvitationEventHandler : MessageProcessor<ApprovedTransferConnectionInvitationEvent>
+    {
+        private readonly EmployerApprenticeshipsServiceConfiguration _employerApprenticeshipsServiceConfiguration;
+
+        public const string UrlFormat = "/accounts/{0}/transfers";
+
+        private readonly EmployerAccountDbContext _db;
+        private readonly ILog _log;
+        private readonly INotificationsApi _notificationsApi;
+
+        public ApprovedTransferConnectionInvitationEventHandler(IMessageSubscriberFactory subscriberFactory, ILog log, EmployerAccountDbContext dbContext, INotificationsApi notificationsApi, EmployerApprenticeshipsServiceConfiguration employerApprenticeshipsServiceConfiguration) : base(subscriberFactory, log)
+        {
+            _db = dbContext;
+            _notificationsApi = notificationsApi;
+            _employerApprenticeshipsServiceConfiguration = employerApprenticeshipsServiceConfiguration;
+            _log = log;
+        }
+
+        protected override async Task ProcessMessage(ApprovedTransferConnectionInvitationEvent messageContent)
+        {
+            var users = _db.Users
```

Comment:
It looks like this query is used in a few places, could it be extracted out into an extension method hanging off `IQueryable<User>`?

Reply:
Agreed - extracted.